src/NuGet.Core/NuGet.Protocol/Resources/PackageMetadataResourceV3.cs

```diff
@@ -38,37 +55,240 @@ public class PackageMetadataResourceV3 : PackageMetadataResource
+        /// <param name="log">Logger Instance.</param>
+        /// <param name="token">Cancellation token.</param>
+        /// <returns></returns>
+        private async Task<RegistrationPage> GetRegistratioIndexPageAsync(
+            HttpSource httpSource,
+            string rangeUri,
+            string packageId,
+            NuGetVersion lower,
+            NuGetVersion upper,
+            HttpSourceCacheContext httpSourceCacheContext,
+            ILogger log,
+            CancellationToken token)
+        {
+            var packageIdLowerCase = packageId.ToLowerInvariant();
+            var registrationPage = await httpSource.GetAsync(
+                            new HttpSourceCachedRequest(
+                                rangeUri,
+                                $"list_{packageIdLowerCase}_range_{lower.ToNormalizedString()}-{upper.ToNormalizedString()}",
+                                httpSourceCacheContext)
+                            {
+                                IgnoreNotFounds = true,
+                            },
+                            async httpSourceResult =>
+                            {
+                                return await ProcessRegistrationIndexAsync<RegistrationPage>(httpSourceResult, token);
+                            },
+                            log,
+                            token);
+
+            return registrationPage;
+        }
+
+        /// <summary>
+        /// Process RegistrationPage
+        /// </summary>
+        /// <param name="registrationPage">Nuget registration page.</param>
+        /// <param name="results">Used to return nuget result.</param>
+        /// <param name="range">Nuget version range.</param>
+        /// <param name="includePrerelease">Whether to include PreRelease versions into result.</param>
+        /// <param name="includeUnlisted">Whether to include Unlisted versions into result.</param>
+        private void ProcessRegistrationPage(
+            RegistrationPage registrationPage,
+            List<PackageSearchMetadataRegistration> results,
+            VersionRange range, bool includePrerelease,
+            bool includeUnlisted,
+            MetadataReferenceCache metadataCache)
+        {
+            foreach (var registrationLeaf in registrationPage.Items)
+            {
+                var catalogEntry = registrationLeaf.CatalogEntry;
+                var version = catalogEntry.Version;
+                var listed = catalogEntry.IsListed;
+
+                if (range.Satisfies(catalogEntry.Version)
+                    && (includePrerelease || !version.IsPrerelease)
+                    && (includeUnlisted || listed))
+                {
+                    //// add in the download url
+                    //if (registrationLeaf.PackageContent != null)
+                    //{
+                    //    catalogEntry.PackageContent = registrationLeaf.PackageContent;
+                    //}
```

Comment:
For me this commented code is dead code someone put for some purpose but never actually done anything.
Because catalogEntry (type of PackageSearchMetadataRegistration) doesn't have property called PackageContent.
It's coming from RegistrationResourceV3.cs>>Line 151.
I guess because that part of code was Json which is not strong type, so it let such useless code exist from the first place. Please let me know if you know anything about this one.

Reply:
I say delete. It's always in git history 😊

Reply:
Deleted.